datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -348,12 +348,17 @@ def _get_setting(name, default):
         # INTERNAL FEATURE, might be removed in future versions
         config['_text_filter_blacklist'] = []
 
-        # Whether or not to use the service account bearer token for authentication
-        # if 'bearer_token_path' is not set, we use /var/run/secrets/kubernetes.io/serviceaccount/token
+        # Whether or not to use the service account bearer token for authentication.
+        # Can be explicitly set to true or false to send or not the bearer token.
+        # If set to the `tls_only` value, the bearer token will be sent only to https endpoints.
+        # If 'bearer_token_path' is not set, we use /var/run/secrets/kubernetes.io/serviceaccount/token
         # as a default path to get the token.
-        config['bearer_token_auth'] = is_affirmative(
-            instance.get('bearer_token_auth', default_instance.get('bearer_token_auth', False))
-        )
+        if instance.get('bearer_token_auth') == 'tls_only' or 'bearer_token_auth' not in instance and default_instance.get('bearer_token_auth') == 'tls_only':
```

Comment:
```suggestion
        bearer_token_auth = _get_setting('bearer_token_auth', False)
        if bearer_token_auth == 'tls_only':
```

And then `is_affirmative(bearer_token_auth)` in the else branch.

Also, can we check the url here too? I think we can avoid having a tri-state `bearer_token_auth` field if we check the url here and set it to true or false based on user settings and the url.

Reply:
Thanks for the suggestion !
It’s clearer like that.